test/ConfigManagerTest.cpp

```diff
@@ -0,0 +1,347 @@
+  bool boolTrueBool;
+  config.createConfigOption<bool>(
+      "boolTrue", "Must be set. Has no default value.", &boolTrueBool);
+  bool boolFalseBool;
+  config.createConfigOption<bool>(
+      "boolFalse", "Must be set. Has no default value.", &boolFalseBool);
+
+  // Add boolean list.
+  std::vector<bool> someBooleanListBoolVector;
+  config.createConfigOption<std::vector<bool>>(
+      "someBooleanList", "Must be set. Has no default value.",
+      &someBooleanListBoolVector);
+
+  // Add string option.
+  std::string myNameString;
+  config.createConfigOption<std::string>(
+      "myName", "Must be set. Has no default value.", &myNameString);
+
+  // Add string list.
+  std::vector<std::string> someStringListStringVector;
+  config.createConfigOption<std::vector<std::string>>(
+      "someStringList", "Must be set. Has no default value.",
+      &someStringListStringVector);
+
+  // Add option with deeper level.
+  std::vector<int> deeperIntVector;
+  config.createConfigOption<std::vector<int>>(
+      {"depth", size_t{0}, "list"}, "Must be set. Has no default value.",
+      &deeperIntVector);
+
+  // This one will not be changed, in order to test, that options, that are
+  // not set at run time, are not changed.
+  int noChangeInt;
+  config.createConfigOption<int>("No_change", "", &noChangeInt, 10);
+
+  // Set those.
+  config.parseConfig(ad_utility::ConfigManager::parseShortHand(
+      R"--(somePositiveNumber : 42, someNegativNumber : -42, someIntegerlist : [40, 41], somePositiveFloatingPoint : 4.2, someNegativFloatingPoint : -4.2, someFloatingPointList : [4.1, 4.2], boolTrue : true, boolFalse : false, someBooleanList : [true, false, true], myName : "Bernd", someStringList : ["t1", "t2"], depth : [{list : [7,8]}])--"));
+
+  // Check, if an option was set to the value, you wanted.
+  auto checkOption =
+      [&config](const auto& content,
+                const ad_utility::ConfigManager::VectorOfKeysForJson& keys) {
+        const auto& option = config.getConfigurationOptionByNestedKeys(keys);
+        ASSERT_TRUE(option.hasSetDereferencedVariablePointer());
+        ASSERT_EQ(content,
+                  option.template getValue<std::decay_t<decltype(content)>>());
+      };
+
+  checkOption(static_cast<int>(42), {"somePositiveNumber"});
+  checkOption(static_cast<int>(-42), {"someNegativNumber"});
+
+  checkOption(std::vector{40, 41}, {"someIntegerlist"});
+
+  checkOption(static_cast<float>(4.2), {"somePositiveFloatingPoint"});
+  checkOption(static_cast<float>(-4.2), {"someNegativFloatingPoint"});
+
+  checkOption(std::vector<float>{4.1, 4.2}, {"someFloatingPointList"});
+
+  checkOption(true, {"boolTrue"});
+  checkOption(false, {"boolFalse"});
+
+  checkOption(std::vector{true, false, true}, {"someBooleanList"});
+
+  checkOption(std::string{"Bernd"}, {"myName"});
+
+  checkOption(std::vector<std::string>{"t1", "t2"}, {"someStringList"});
+
+  checkOption(std::vector{7, 8}, {"depth", size_t{0}, "list"});
```

Comment:
Same here, check the values of the underlying storage.

Reply:
Same here, I do.

Reply:
Same here, I don't think you do (okay, you do, but you still need to do it explicitly to have a proper test that tests the externally visible behavior).

Reply:
If you really want to, you can give the pointers to your outside variables as additional arguments to `checkOption`, maybe that is a compromise).

Reply:
Done, see [c7013fa](https://github.com/ad-freiburg/qlever/pull/977/commits/c7013fafa60f33ccff75a739a51ce993f8b1fa43).